src/Approximations/overapproximate.jl

```diff
@@ -516,3 +516,73 @@ function overapproximate(cap::Intersection{N,
     return overapproximate(swap(cap), dir; kwargs...)
 end
 
+"""
+    overapproximate(lm::LinearMap{N, <:CartesianProductArray{N,
+                            <:LazySet{N}}}, ::Type{<:CartesianProductArray}, oa=Hyperrectangle)  where {N}
+
+Overapproximating a lazy linear map of cartesian product array with template directions for each block.
+
+### Input
+
+- `lm`   -- lazy linear map of cartesian product array
+- Type{<:CartesianProductArray} -- type to specify decomposed overapproximation
+- `oa` -- (concrete) direction representation
+
+### Output
+
+An `CartesianProductArray` with overapproximation of the each block with the directions from `dir`.
+"""
+function overapproximate(lm::LinearMap{N, <:CartesianProductArray{N,
+                            <:LazySet{N}}}, ::Type{<:CartesianProductArray}, oa=Hyperrectangle) where {N}
+
+    @assert size(lm.M, 2) == dim(lm.X) "matrix needs to be commensurate with the cartesian product"
+
+    cp = lm.X
+    M = lm.M
+
+    col_end_ind = 0
+
+    return decomposed_overapproximation(cp, M, col_end_ind, oa)
+end
+
+#template directions
+function decomposed_overapproximation(cpa::CartesianProductArray{N,<:LazySet{N}},
+                                      M::AbstractMatrix{N},
+                                      col_end_ind::Int, oa) where {N}
+    array = allocate_result(oa, N)
+    sizehint!(array,length(cpa.array))
+
+    for i in 1:size(M, 2)
+        ms = overapproximate_row_blocks(cpa, M, i)
```

Comment:
You cannot iterate over all integers. You need to iterate over all blocks. In general the blocks are not necessarily one-dimensional.
I am surprised that this was not detected by the test.